File: mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegView.java
package com.github.niqdev.mjpeg;

import android.view.SurfaceView;

public interface MjpegView {

    void setSource(MjpegInputStream stream);

    void setDisplayMode(DisplayMode mode);

    void showFps(boolean show);

    void flipSource(boolean flip);

    void flipHorizontal(boolean flip);

    void flipVertical(boolean flip);

    void stopPlayback();

    boolean isStreaming();

    void setResolution(int width, int height);

    void freeCameraMemory();

    void setOnFrameCapturedListener(OnFrameCapturedListener onFrameCapturedListener);

    void setCustomBackgroundColor(int backgroundColor);

    void setFpsOverlayBackgroundColor(int overlayBackgroundColor);

    void setFpsOverlayTextColor(int overlayTextColor);

    SurfaceView getSurfaceView();

    void resetTransparentBackground();

    void setTransparentBackground();

    void clearStream();
}

File: mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegView.kt
package com.github.niqdev.mjpeg

import android.view.SurfaceView

interface MjpegView {
    fun setSource(stream: MjpegInputStream)
    fun setDisplayMode(mode: DisplayMode)
    fun showFps(show: Boolean)
    fun flipSource(flip: Boolean)
    fun flipHorizontal(flip: Boolean)
    fun flipVertical(flip: Boolean)
    fun setRotate(degrees: Float)
    fun stopPlayback()
    val isStreaming: Boolean
    fun setResolution(width: Int, height: Int)
    fun freeCameraMemory()
    fun setOnFrameCapturedListener(onFrameCapturedListener: OnFrameCapturedListener)
    fun setCustomBackgroundColor(backgroundColor: Int)
    fun setFpsOverlayBackgroundColor(overlayBackgroundColor: Int)
    fun setFpsOverlayTextColor(overlayTextColor: Int)
    val surfaceView: SurfaceView
    fun resetTransparentBackground()
    fun setTransparentBackground()
    fun clearStream()
}
